YololCompetition/Services/Schedule/InMemoryScheduler.cs

```diff
@@ -56,24 +58,50 @@ public async Task Start()
                 // If there is no challenge running try to start a new one
                 if (current == null)
                 {
-                    Console.WriteLine("There is no current challenge - attempting to start a new one");
+                    Console.WriteLine("There is no current challenge - checking if challenge finished within the last 24 hours");
 
-                    // Start the next challenge, if there isn't one wait a while
-                    var next = await _challenges.StartNext();
-                    if (next == null)
+                    var currentTime = DateTime.UtcNow;
+                    var startTime = (currentTime.Date).AddMinutes(_config.ChallengeStartTime);  //Gets the time today that the challenge would be starting if it were starting today
+                    var searchTime = currentTime.AddHours(-24);
+                    var recent = await _challenges.GetChallengesByEndTime(searchTime.UnixTimeStamp());
+
+                    if (recent == null)
                     {
-                        Console.WriteLine("No challenges available, waiting for a while...");
-                        State = SchedulerState.WaitingNoChallengesInPool;
-                        EndTime = null;
-                        await Task.Delay(TimeSpan.FromMinutes(1));
-                        continue;
+
+                        // Start the next challenge, if there isn't one wait a while
+                        var next = await _challenges.StartNext();
```

Comment:
If 24 hours have elapsed since the last challenge ended this starts a new one immediately, instead of waiting for the next start time.

Reply:
This is intentional. I was thinking that challenge uptime is more important than that the challenge start at the exact right time.

Reply:
This discussion is the one I'm waiting on before committing the other changes. 

The changes to the challenge start next is directly related to this decision. The start time used in the start next is based on UTC, and UTC wise we are likely to start in the final 7 hours of the day (sometime between 1700 UTC and 2000 UTC imo). Since the start time + duration is the start time of "today" + duration, if the current time when starting the challenge is after the start time of day, then the duration of the challenge will be slightly less. If it is earlier in the day than the start time then the duration will be longer. This likely gives us a range of 65 hours to 92 hours, with the bot automatically fixing the schedule to be based around the configured start time and duration for future challenges.

Reply:
I think it's reasonable to leave it as it is. If one starts early (due to the very rare event that Referee was down for >24 hours) the one _after_ that will start on time and fix the schedule so overall it's really not worth worrying about.